packages/backend/src/peripherals/database/EventRepository.ts

```diff
@@ -72,6 +76,38 @@ export class EventRepository extends BaseRepository {
     return rows.map(toRecord)
   }
 
+  async getAggregatedByProjectAndGranularity(
+    projectId: ProjectId,
+    granularity: 'hour' | 'day',
+    from: UnixTime = new UnixTime(0),
+  ): Promise<EventRecordAggregated[]> {
+    const knex = await this.knex()
+    const rows = (await knex('events')
+      .where('project_id', projectId.toString())
+      .where('unix_timestamp', '>=', from.toDate())
+      .select(
+        knex.raw(
+          `date_trunc('${granularity}', unix_timestamp) AS unix_timestamp`,
```

Comment:
Strings should never be concatenated into SQL because this leaves a possibility for SQL injection vulnerabilities. Use bindings instead

Reply:
Yes. Also, feel free to migrate to simply using .raw as @michalsidzej did: https://github.com/l2beat/l2beat/pull/634/files#diff-4fe439c9479ba1dbaf66a5452361b2095a713399847cf7f72e0457f3da6127d9R62